src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java

```diff
@@ -134,13 +153,56 @@ private MongoClientOptions createOptions() {
         if (threadsAllowedToBlockForConnectionMultiplier != null) {
             optionsBuilder.threadsAllowedToBlockForConnectionMultiplier(threadsAllowedToBlockForConnectionMultiplier);
         }
-        if (enableSSL != null) {
-            optionsBuilder.sslEnabled(enableSSL);
+        SSLContext sslContext = getSSLContext();
+        if (sslContext == null) {
+            if (enableSSL != null) {
+                optionsBuilder.sslEnabled(enableSSL);
+                if (sslInvalidHostNameAllowed != null) {
+                    optionsBuilder.sslInvalidHostNameAllowed(sslInvalidHostNameAllowed);
+                }
+            }
+        } else {
+            optionsBuilder.sslEnabled(true);
             if (sslInvalidHostNameAllowed != null) {
                 optionsBuilder.sslInvalidHostNameAllowed(sslInvalidHostNameAllowed);
             }
+            optionsBuilder.sslContext(sslContext);
+        }
+        return optionsBuilder;
+    }
+
+    private SSLContext getSSLContext() throws SchedulerConfigException {
```

Comment:
IMHO it would be good to move it to a `SSLContextFactory` or some such to make this class smaller.

Reply:
Like a helper class with static methods?
Sure, why not.

Besides this, do you agree with the principle @pwojnowski ?
Are you able to merge pull requests or should we wait for @michaelklishin's review?

Thanks

Reply:
Basically it looks good to me. I'm just wondering how to test that - at least manually. :thinking: 

Yes, I can merge PRs.

Reply:
Bette if it wouldn't be static as they are harder to test. It could be a simple class, instantiated in MongoConnectorBuilder. Having a separate class it will be easy to move it around if needed.

Reply:
To test this, I'm afraid that the only solution is to configure a MongoDB/SSL connection with a keystore/truststore...
To be honest, we haven't done it but relied on a customer testing the patch.

OK, I will move the code to a separate class instantiated in `MongoConnectorBuilder`.

Reply:
Hi, I've just pushed the suggested change: externalizing the SSL logic to an `SSLContextFactory` class.
Sorry it took so long.